### PR TITLE
JBPM 5962 - Adding support to Control Point Handler to be extended via Factory

### PR DESCRIPTION
**Pull request by tiagodolphine**
@romartin @Josephblt  @mbiarnes porting the PR to the Lienzo kiegroup repo.

----

Add support to Wires Connector Control Points Handlers to be extended and injected on WiresManager through the WiresHandlerFactory, this will be used by Stunner to integrate with the Model and Commands.
Some methods were added to WiresConnectorControlImpl to create and remove control points from connectors.
Code refactory improvements based on factory to WiresShapeHandler, in the same way as for connectors on WiresHandlerFactory.

Related JIRA https://issues.jboss.org/browse/JBPM-5962.

Some tests were added on the https://github.com/kiegroup/lienzo-tests/pull/4 as well.

@romartin can you please help to review and add any other that may be important on the reviewing .

Thanks!

### CODE REVIEW

**Comment by romartin** [comment]
Jenkins execute full downstream build

File: src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/WiresShapeHandlerImpl.java [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+/**
+ * This handler's goals are:
+ * - Delegate some mouse interactions for a single wires shape to its wires shape control instance
+ * - Displays some highlights to provide feedback for containment and docking operations.
+ */
+public class WiresShapeHandlerImpl
+        extends WiresManager.WiresDragHandler
+        implements
+        WiresShapeHandler {
```
Comment (LuboTerifaj) [inline src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/WiresShapeHandlerImpl.java]: Why is handler on new line?

File: src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/WiresConnectorControl.java [hunk 2]
```diff
@@ -16,7 +18,28 @@
...
     void move(double dx, double dy, boolean midPointsOnly, boolean moveLinePoints);
 
-    void addControlPoint( double x, double y );
+    /**
+     * Add a control point on the existing {@link WiresConnector#getLine()} returning the selected index to the control point
+     * @param x position
+     * @param y position
+     * @return index to the added Control Point
+     */
+    int addControlPoint( double x, double y );
```
Comment (LuboTerifaj) [inline src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/WiresConnectorControl.java]: I would delete spaces in new code before and after formal parameters as it is in other methods.

File: src/main/java/com/ait/lienzo/client/core/shape/wires/SelectionManager.java [hunk 3]
```diff
@@ -339,6 +339,15 @@ private boolean selectionEventHandlingComplete(MouseEvent<? extends EventHandler
         }
     }
 
+    public void rebuildSelectionArea() {
+        if (m_selected.isEmpty())
+        {
```
Comment (LuboTerifaj) [inline src/main/java/com/ait/lienzo/client/core/shape/wires/SelectionManager.java]: I would use one approach of putting brackets in same method.

**Pull request merged**